lmcache/experimental/storage_backend/abstract_backend.py

```diff
@@ -0,0 +1,91 @@
+import abc
+from concurrent.futures import Future
+from typing import Optional
+
+import torch
+
+from lmcache.experimental.memory_management import BufferMemoryObj, MemoryObj
+from lmcache.utils import CacheEngineKey
+
+
+class StorageBackendInterface(metaclass=abc.ABCMeta):
+
+    def __init__(
+        self,
+        dst_device: str = "cuda",
+    ):
+        """
+        Initialize the storage backend. 
+
+        :param dst_device: the device where the blocking retrieved KV is stored,
+            could be either "cpu", "cuda", or "cuda:0", "cuda:1", etc.
+
+        :raise: RuntimeError if the device is not valid
+        """
+        try:
+            torch.device(dst_device)
+        except RuntimeError:
+            raise
+
+        self.dst_device = dst_device
+
+    @abc.abstractmethod
+    def contains(self, key: CacheEngineKey) -> bool:
+        """
+        Check whether key is in the storage backend. 
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def insert_key(self, key: CacheEngineKey, size: int):
+        """
+        Insert the keu after data is put to storage backend.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def submit_put_task(self, key: CacheEngineKey, obj: MemoryObj) -> Future:
+        """
+        An async function to put the MemoryObj into the storage backend.
+
+        :param CacheEngineKey key: The key of the MemoryObj.
+        :param MemoryObj obj: The MemoryObj to be stored.
+        
+        :return: a future object
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def submit_prefetch_task(
+        self,
+        key: CacheEngineKey,
+    ) -> Optional[Future]:
+        """
+        An async function to get the MemoryObj from the storage backend.
+
+        :param CacheEngineKey key: The key of the MemoryObj.
+
+        :return: a future object
```

Comment:
In what case it will return None? Please add a sentence to explain it a bit in the docstring